packages/ag-charts-enterprise/src/features/color-picker/colorPicker.ts

```diff
@@ -91,8 +91,10 @@ export class ColorPicker extends AnchoredPopover<ColorPickerOptions> {
                 colorInput.value = colorString.toUpperCase();
             }
 
-            const plainColor = Color.fromHSB(h, s, v, 1).toHexString();
-            opts.onChange?.(colorString, plainColor, a);
+            if (trackChange || opts.color == null) {
+                const plainColor = Color.fromHSB(h, s, v, 1).toHexString();
+                opts.onChange?.(colorString, plainColor, a);
+            }
```

Comment:
Previously, this would always fire the onChange when it immediately opened creating a ghost action in the history manager for undo/redo.